src/ui/public/utils/get_vislib_options.js

```diff
@@ -0,0 +1,14 @@
+import _ from 'lodash';
+
+export default function getVislibOptions(vis) {
+  return _.assign(
+    {},
+    vis.type.params.defaults,
+    {
+      type: vis.type.name,
+      // Add attribute which determines whether an index is time based or not.
+      hasTimeField: vis.indexPattern && vis.indexPattern.hasTimeField()
+    },
+    vis.params
+  );
+}
```

Comment:
I think I'm not quite getting the meaning behind this object. Based on how it's used, it seems like this is an aggregation of all of the properties related to a visualization which can be changed by the user, and are therefore saveable, right?